### ISSUE TITLE
Ability to add a new certfp without having to use the cert

### ISSUE THREAD

**Issue by PeGaSuS-Coder** [anope/anope#326]
**Description**
Currently, we need to be using the certificate in order to add it to our NickServ certificate list.
It would be nice if we could add another certfp without having to be using the actual certificate.

**Why this would be useful**
This would be useful for change certificate and certfp, reconnect and get automatically identified.
Otherwise we need to change cert, reconnect, add new certfp and reconnect again. That way we'd only need to reconnect once.

Cheers

**Comment by SadieCat** [comment]
The problem with this is that:

1. Fingerprints are public info so you can trick people to log into an account by adding their cert to an account.
2. A certificate can only be added to one account so allowing this makes it so you can block people from adding their cert to an account.

**Comment by Techman** [comment]
I agree with Sadie on this. Social engineering risk should be taken seriously.

It is not much effort to add additional certificates using the current system.

**Comment by SadieCat** [comment]
I think we could maybe do something more user friendly by having new certs in a "pending" state that a user has to use within a short period from the same host in order for them to be added?

**Comment by PeGaSuS-Coder** [comment]
You mean like being able to add it but we need to connect to the network in the next X minutes for it to become active?

**Comment by henk84** [comment]
> I think we could maybe do something more user friendly by having new certs in a "pending" state that a user has to use within a short period from the same host in order for them to be added?

Since this would just be an enhancement, IMHO this should not be implemented with this inherent security risk, even if it is mitigated by forcing an attacker to be more accurate in their timing.

The only alternative I can think of, though: the user will get a notification that they have connected using a cert they have not yet verified and ask them to enter their password to verify it.
On the other hand this does not seem to have any advantage over the way it is currently done.

**Comment by TehPeGaSuS** [comment]
> Since this would just be an enhancement, IMHO this should not be implemented with this inherent security risk, even if it is mitigated by forcing an attacker to be more accurate in their timing.

Not only it's timed but you also need to connect from the same host with the matching fingerprint which is more difficult to do.
